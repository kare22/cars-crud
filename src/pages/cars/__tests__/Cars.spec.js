import { createLocalVue, mount } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Cars from '@/pages/cars/index.vue';

import testData from './test-data';

jest.mock('@/store');

describe('Cars', () => {
    const mocks = {};

    let actions;
    let getters;
    let store;

    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    const router = new VueRouter();

    function getComponent(options = {}) {
        return mount(Cars, {
            ...options,
            mocks,
            store,
            localVue,
            router,
        });
    }

    beforeEach(() => {
        actions = {
            'cars/get': () => [],
            'categories/get': () => [],
            'probabilities/get': () => [],
            'impacts/get': () => [],
        };
        getters = {
            'cars/categorized': () => testData.cars,
            'categories/data': () => testData.categories,
            'impacts/data': () => testData.impacts,
            'probabilities/data': () => testData.probabilities,
        };
        store = new Vuex.Store({
            actions,
            getters,
        });
    });

    it('snapshot test', () => {
        expect(getComponent()).toMatchSnapshot();
    });

    it('should show all categories', async() => {
        const wrapper = await getComponent({});

        testData.categories.forEach(category => {
            expect(wrapper.find(`#category_${category.id}`).exists()).toBeTruthy();
        });
    });

    it('should have the right amount of cars', async() => {
        const wrapper = await getComponent({});

        testData.categories.forEach(category => {
            const cars = testData.cars[category.id] || [];

            cars.forEach(car => {
                expect(wrapper.find(`#category_${category.id}`).find(`#car_${car.id}`).exists()).toBeTruthy();
            });
        });
    });
});

