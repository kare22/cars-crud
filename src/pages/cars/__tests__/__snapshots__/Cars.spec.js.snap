// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cars snapshot test 1`] = `
<div class="mx-3">
  <div class="row">
    <div class="col-4 text-left"><button class="btn btn-info">
        New Car
      </button></div>
    <div class="col-4">
      <h1 class="text-info font-weight-bold">
        Car details
      </h1>
    </div>
    <div class="col-4"></div>
  </div>
  <div id="category_1">
    <div>
      <b-card-header header-tag="header" role="tab" class="p-1">
        <b-button block="" variant="info">
          <div class="d-flex justify-content-between">
            <div>
              Category 1 (3)
            </div>
            <div>
              <biconchevrondown></biconchevrondown>
            </div>
          </div>
        </b-button>
      </b-card-header>
      <b-collapse id="collapse-1">
        <b-card>
          <div>
            <div id="car_1">
              <validationobserver></validationobserver>
            </div>
            <div class="separator border-info"></div>
          </div>
          <div>
            <div id="car_2">
              <validationobserver></validationobserver>
            </div>
            <div class="separator border-info"></div>
          </div>
          <div>
            <div id="car_3">
              <validationobserver></validationobserver>
            </div>
            <div class="separator border-info"></div>
          </div>
        </b-card>
      </b-collapse>
    </div>
  </div>
  <div id="category_2">
    <div>
      <b-card-header header-tag="header" role="tab" class="p-1">
        <b-button block="" variant="info">
          <div class="d-flex justify-content-between">
            <div>
              Category 2 (0)
            </div>
            <div>
              <biconchevrondown></biconchevrondown>
            </div>
          </div>
        </b-button>
      </b-card-header>
      <b-collapse id="collapse-2">
        <b-card></b-card>
      </b-collapse>
    </div>
  </div>
  <div id="category_3">
    <div>
      <b-card-header header-tag="header" role="tab" class="p-1">
        <b-button block="" variant="info">
          <div class="d-flex justify-content-between">
            <div>
              Category 3 (1)
            </div>
            <div>
              <biconchevrondown></biconchevrondown>
            </div>
          </div>
        </b-button>
      </b-card-header>
      <b-collapse id="collapse-3">
        <b-card>
          <div>
            <div id="car_4">
              <validationobserver></validationobserver>
            </div>
            <div class="separator border-info"></div>
          </div>
        </b-card>
      </b-collapse>
    </div>
  </div>
  <!---->
</div>
`;
